lib/features/manage_account/presentation/mailbox_visibility/mailbox_visibility_controller.dart

```diff
@@ -0,0 +1,207 @@
+import 'package:tmail_ui_user/features/mailbox/domain/usecases/refresh_all_mailbox_interactor.dart';
+import 'package:tmail_ui_user/features/mailbox/domain/usecases/subscribe_mailbox_interactor.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_categories.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_categories_expand_mode.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_node.dart';
+import 'package:tmail_ui_user/features/manage_account/presentation/manage_account_dashboard_controller.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+import 'package:tmail_ui_user/main/routes/route_navigation.dart';
+
+class MailboxVisibilityController extends BaseMailboxController {
+  GetAllMailboxInteractor? _getAllMailboxInteractor;
+  RefreshAllMailboxInteractor? _refreshAllMailboxInteractor;
+  SubscribeMailboxInteractor? _subscribeMailboxInteractor;
+  final _accountDashBoardController = Get.find<ManageAccountDashBoardController>();
+  final _appToast = Get.find<AppToast>();
+  final _imagePaths = Get.find<ImagePaths>();
+  final _responsiveUtils = Get.find<ResponsiveUtils>();
+  final mailboxCategoriesExpandMode = MailboxCategoriesExpandMode.initial().obs;
+  final mailboxListScrollController = ScrollController();
+
+  Map<Role, MailboxId> mapDefaultMailboxIdByRole = {};
+  Map<MailboxId, PresentationMailbox> mapMailboxById = {};
+  PresentationMailbox? outboxMailbox;
+
+  jmap.State? _currentMailboxState;
+
+  MailboxVisibilityController(
+    treeBuilder,
+  ) : super(treeBuilder);
+
+  @override
+  void onInit() {
+    super.onInit();
+    try {
+      _getAllMailboxInteractor = Get.find<GetAllMailboxInteractor>();
+      _refreshAllMailboxInteractor = Get.find<RefreshAllMailboxInteractor>();
+      _subscribeMailboxInteractor = Get.find<SubscribeMailboxInteractor>();
+    } catch (e) {
+      logError('MailboxVisibilityController::onInit(): ${e.toString()}');
+    }
+  }
+
+  @override
+  void onDone() {
+    viewState.value.fold((failure) {}, (success) {
+      if (success is GetAllMailboxSuccess) {
+        _currentMailboxState = success.currentMailboxState;
+        _buildMailboxTreeHasSubscribed(success.mailboxList);
+      } else if (success is RefreshChangesAllMailboxSuccess) {
+        _currentMailboxState = success.currentMailboxState;
+        _refreshMailboxTreeHasSubscribed(success.mailboxList);
+      } else if (success is SubscribeMailboxSuccess) {
+        _subscribeMailboxSuccess(success);
+      }
+    });
+
+  }
+
+  @override
+  void onReady() {
+    final _session = _accountDashBoardController.sessionCurrent.value;
+    final _accountId = _accountDashBoardController.accountId.value;
+    if(_session != null && _accountId != null) {
+      getAllMailboxAction(_session, _accountId);
+    }
+    _initCollapseMailboxCategories();
+    super.onReady();
+  }
+
+  void _initCollapseMailboxCategories() {
```

Comment:
why we need `collapse`  mailbox categories?

Reply:
srr, I'm using redundant here